src/com/palmergames/bukkit/towny/utils/SpawnUtil.java

```diff
@@ -0,0 +1,395 @@
+package com.palmergames.bukkit.towny.utils;
+
+import java.util.List;
+
+import org.bukkit.Chunk;
+import org.bukkit.Location;
+import org.bukkit.entity.Player;
+import org.bukkit.event.player.PlayerTeleportEvent.TeleportCause;
+
+import com.earth2me.essentials.Teleport;
+import com.earth2me.essentials.User;
+import com.palmergames.bukkit.towny.Towny;
+import com.palmergames.bukkit.towny.TownyAPI;
+import com.palmergames.bukkit.towny.TownyEconomyHandler;
+import com.palmergames.bukkit.towny.TownyMessaging;
+import com.palmergames.bukkit.towny.TownySettings;
+import com.palmergames.bukkit.towny.TownyTimerHandler;
+import com.palmergames.bukkit.towny.TownyUniverse;
+import com.palmergames.bukkit.towny.exceptions.EconomyException;
+import com.palmergames.bukkit.towny.exceptions.TownyException;
+import com.palmergames.bukkit.towny.object.Nation;
+import com.palmergames.bukkit.towny.object.NationSpawnLevel;
+import com.palmergames.bukkit.towny.object.Resident;
+import com.palmergames.bukkit.towny.object.SpawnType;
+import com.palmergames.bukkit.towny.object.Town;
+import com.palmergames.bukkit.towny.object.TownBlock;
+import com.palmergames.bukkit.towny.object.TownSpawnLevel;
+import com.palmergames.bukkit.towny.object.TownyEconomyObject;
+import com.palmergames.bukkit.towny.object.TownyObject;
+import com.palmergames.bukkit.towny.permissions.PermissionNodes;
+import com.palmergames.bukkit.towny.tasks.CooldownTimerTask;
+import com.palmergames.bukkit.towny.tasks.CooldownTimerTask.CooldownType;
+
+public class SpawnUtil {
+
+	private static Towny plugin;
+
+	public static void initialize(Towny plugin) {
+		SpawnUtil.plugin = plugin;
+	}
+
+	/**
+	 * Central Util for /res, /t, /n, /ta spawn commands.
+	 * 
+	 * @param player       - Player using spawn command.
+	 * @param split        - Remaining command arguments, used primarily for
+	 *                     outposts.
+	 * @param townyObject  - Either a town or nation depending on source command.
+	 * @param notAffordMSG - Message shown when a player cannot afford their
+	 *                     teleport.
+	 * @param outpost      - Whether this is an outpost or not.
+	 * @param spawnType    - SpawnType.RESIDENT/TOWN/NATION
+	 * @throws TownyException - Thrown if any of the vital conditions are not met.
+	 */
+	public static void sendToTownySpawn(Player player, String[] split, TownyObject townyObject, String notAffordMSG, boolean outpost, SpawnType spawnType) throws TownyException {

```

Comment:
Rename param `split` to something like `args` to make the usage clearer throughout the method. `split` is more of a name for a temporary string split variable.

Reply:
In the context of towny it makes the most sense though, just about every time we name it split.